### PR TITLE
Add ClassStructureSniff

### PR DESCRIPTION
**Pull request by grongor**
Part of https://github.com/slevomat/coding-standard/issues/701 series.

Fixes https://github.com/slevomat/coding-standard/issues/624

:wrench:  TODO: add readme if sniff is accepted

### CODE REVIEW

File: SlevomatCodingStandard/Sniffs/Classes/ClassStructureSniff.php [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+
+		$file->fixer->endChangeset();
+	}
+
+	private function findPreviousMemberEndPointer(File $file, int $memberPointer): int
+	{
+		$endTypes = [T_OPEN_CURLY_BRACKET, T_CLOSE_CURLY_BRACKET, T_SEMICOLON];
+		$previousMemberEndPointer = TokenHelper::findPrevious($file, $endTypes, $memberPointer - 1);
+		if ($previousMemberEndPointer === null) {
+			throw new RuntimeException('Previous member end pointer not found');
```
Comment (kukulich) [inline SlevomatCodingStandard/Sniffs/Classes/ClassStructureSniff.php]: Are we able to remove all `throw new RuntimeException` ?

Comment (grongor) [reply SlevomatCodingStandard/Sniffs/Classes/ClassStructureSniff.php]: Sure, I'll replace it with asserts.

**Comment by kukulich** [comment]
LGTM.

**Comment by kukulich** [comment]
@grongor How can I set different order in my ruleset? Can you please post an example?

**Comment by kukulich** [comment]
@grongor It looks it does not work at all :) I've already refectored (and fixed) it on my localhost :)

**Comment by grongor** [comment]
Sorry for the late response ... I'm sorry, I never actually tested the settings outside the tests ... I think it worked, just not with the constants. Thanks for the fix.

**Comment by grongor** [comment]
There is a little issue with your changes: you can't set multiple groups to have the same expected order (ergo be ignored by the check). That might've been useful for people who wanted to check "just that constants are before properties" or something like that. Also, if you don't specify some groups in the custom config then they will be expected to be at the end of the class, which doesn't make much sense to me.

I personally don't care about these issues, I just wanted to point out that they now exist.

**Comment by kukulich** [comment]
@grongor You can :) See readme and example https://github.com/slevomat/coding-standard#slevomatcodingstandardclassesclassstructure-

**Comment by grongor** [comment]
Oh nice, I just tried to figure it out from the code and it seemed at the first glance that it's not possible. Perfect then! Missing groups will still be at the end, but I guess that's ok :)

**Comment by kukulich** [comment]
Yeah, I think you can just specify all groups and you're ok :)

**Pull request merged**